File: app/src/main/java/com/buel/holyhelpers/view/BriefingViewPager.java
package com.buel.holyhelpers.view;

import android.content.Context;
import android.view.View;

import com.buel.holyhelpers.R;
import com.buel.holyhelpers.data.CommonData;
import com.buel.holyhelpers.view.viewPage.CreateViewPageAdater;
import com.buel.holyhelpers.view.viewPage.briefingPageViewFragment.AbstBriefingFragment;
import com.commonLib.Common;
import com.commonLib.MaterialDailogUtil;
import com.google.android.material.tabs.TabLayout;
import com.orhanobut.logger.LoggerHelper;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

/**
 * Created by 19001283 on 2018-07-19.
 */

public class BriefingViewPager  {

    private static ViewPager viewPager = null;
    private static FragmentPagerAdapter pagerAdapter = null;
    private static Context context = null;

    public static void setMainViewPage(Context context, int pageAdapterNum) {

        BriefingViewPager.context = context;
        View rootView = Common.getRootView(context);
        // Find the view pager that will allow the user to swipe between fragments

        viewPager = rootView.findViewById(R.id.mainActivity_viewpager);
        // Create an adapter that knows which fragment should be shown on each page

        pagerAdapter = CreateViewPageAdater.onCreateFragmentAdapter(context, ((AppCompatActivity) context).getSupportFragmentManager(), pageAdapterNum);

        // Set the adapter onto the view pager
        viewPager.setAdapter(pagerAdapter);

        // Give the TabLayout the ViewPager
        TabLayout tabLayout = rootView.findViewById(R.id.mainActivity_tablayout);
        tabLayout.setupWithViewPager(viewPager);
        viewPager.invalidate();

        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                //viewPager.getAdapter().notifyDataSetChanged();
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        pagerAdapter.notifyDataSetChanged();
    }

    public static void setDataAndTime(MaterialDailogUtil.OnDialogSelectListner dialogSelectListner) {
        MaterialDailogUtil.Companion.datePickerDialog(
                context,
                s -> {

                    LoggerHelper.d(CommonData.getSelectedYear() + "/" +
                            CommonData.getSelectedMonth() + "/" +
                            CommonData.getSelectedDay() + "/" +
                            CommonData.getSelectedDayOfWeek());

                    MaterialDailogUtil.Companion.showSingleChoice(
                            context,
                            R.array.days_option,
                            s1 -> {
                                CommonData.setSelectedDays(Integer.parseInt(s1));
                                LoggerHelper.d("setDataAndTime OnDialogSelectListner" ,"CommonData.getSelectedDays() : " + CommonData.getSelectedDays());

                                Fragment fragment = Common.getCurrentFragment(viewPager, pagerAdapter);
                                pagerAdapter.notifyDataSetChanged();
                                ((AbstBriefingFragment) fragment).getAttandData();
                                dialogSelectListner.onSelect("onComplete");
                            });
                }
        );
    }


    public static ViewPager getViewPager() {
        return viewPager;
    }

    public static void setViewPager(ViewPager viewPager) {
        BriefingViewPager.viewPager = viewPager;
    }

    public static FragmentPagerAdapter getPagerAdapter() {
        return pagerAdapter;
    }

    public static void setPagerAdapter(FragmentPagerAdapter pagerAdapter) {
        BriefingViewPager.pagerAdapter = pagerAdapter;
    }
}

File: app/src/main/java/com/buel/holyhelpers/view/BriefingViewPager.kt
package com.buel.holyhelpers.view

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.buel.holyhelpers.R
import com.buel.holyhelpers.data.CommonData
import com.buel.holyhelpers.data.CommonData.selectedDay
import com.buel.holyhelpers.data.CommonData.selectedDayOfWeek
import com.buel.holyhelpers.data.CommonData.selectedDays
import com.buel.holyhelpers.data.CommonData.selectedYear
import com.buel.holyhelpers.view.viewPage.CreateViewPageAdater
import com.buel.holyhelpers.view.viewPage.briefingPageViewFragment.AbstBriefingFragment
import com.commonLib.Common
import com.commonLib.MaterialDailogUtil.Companion.datePickerDialog
import com.commonLib.MaterialDailogUtil.Companion.showSingleChoice
import com.commonLib.MaterialDailogUtil.OnDialogSelectListner
import com.google.android.material.tabs.TabLayout
import com.orhanobut.logger.LoggerHelper

/**
 * Created by 19001283 on 2018-07-19.
 */
object BriefingViewPager {
    private lateinit var viewPager: ViewPager
    private lateinit var pagerAdapter: FragmentPagerAdapter
    private var context: Context? = null
    fun setMainViewPage(context: Context, pageAdapterNum: Int) {
        BriefingViewPager.context = context
        val rootView = Common.getRootView(context)
        // Find the view pager that will allow the user to swipe between fragments
        viewPager = rootView.findViewById(R.id.mainActivity_viewpager)
        // Create an adapter that knows which fragment should be shown on each page
        pagerAdapter = CreateViewPageAdater.onCreateFragmentAdapter(context, (context as AppCompatActivity).supportFragmentManager, pageAdapterNum)
        // Set the adapter onto the view pager
        viewPager.setAdapter(pagerAdapter)
        // Give the TabLayout the ViewPager
        val tabLayout: TabLayout = rootView.findViewById(R.id.mainActivity_tablayout)
        tabLayout.setupWithViewPager(viewPager)
        viewPager.invalidate()
        viewPager.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
            override fun onPageSelected(position: Int) { //viewPager.getAdapter().notifyDataSetChanged();
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
        pagerAdapter.notifyDataSetChanged()
    }

    fun setDataAndTime(dialogSelectListner: OnDialogSelectListner) {
        datePickerDialog(
                context!!,
                object : OnDialogSelectListner {
                    override fun onSelect(s: String) {
                        LoggerHelper.d(""+ selectedYear + "/" +
                                CommonData.selectedMonth + "/" +
                                selectedDay + "/" +
                                selectedDayOfWeek)
                        showSingleChoice(
                                context!!,
                                R.array.days_option,
                                object : OnDialogSelectListner {
                                    override fun onSelect(s: String) {
                                        selectedDays = s.toInt()
                                        LoggerHelper.d("setDataAndTime OnDialogSelectListner", "CommonData.getSelectedDays() : $selectedDays")
                                        val fragment = Common.getCurrentFragment(viewPager, pagerAdapter)
                                        pagerAdapter!!.notifyDataSetChanged()
                                        (fragment as AbstBriefingFragment).getAttandData()
                                        dialogSelectListner.onSelect("onComplete")
                                    }
                                })
                    }
                }
        )
    }

    fun getViewPager(): ViewPager? {
        return viewPager
    }

    fun setViewPager(viewPager: ViewPager?) {
        BriefingViewPager.viewPager = viewPager!!
    }

    fun getPagerAdapter(): FragmentPagerAdapter? {
        return pagerAdapter
    }

    fun setPagerAdapter(pagerAdapter: FragmentPagerAdapter?) {
        BriefingViewPager.pagerAdapter = pagerAdapter!!
    }
}